math/randomnessGenerator.go

```diff
@@ -0,0 +1,37 @@
+package math
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math/rand"
+)
+
+type SeedRandReader struct {
```

Comment:
make unexported & add tests.
Please do add tests that check that for a predefined string we always get the same n "random" bytes. In this way we have the certainty that we won't add backwards compatibility bugs. (the predefined strings should have some comments stating that those strings should not change)